src/liboslexec/llvm_gen.cpp

```diff
@@ -187,6 +187,13 @@ BackendLLVM::llvm_run_connected_layers(Symbol& sym, int symindex, int opnum,
         const Connection& con(inst()->connection(c));
         // If the connection gives a value to this param
         if (con.dst.param == symindex) {
+            // Non-lazy layers are run upfront directly via llvm_call_layer.
+            // Eliding them here doesn't change the semantics of execution,
+            // but it will prevent optixTrace calls from being repeatedly
+            // inlined when lazytrace=0.
+            if (!group()[con.srclayer]->run_lazily())
+                continue;
```

Comment:
This is the only part of the PR that affects the status quo lazytrace=1 path. I could narrow it to only execute when lazytrace=0, but it seems better to always skip the run-layer check on unconditionally-executed layers.